### PR TITLE
 informative error message on /api/authenticate/baseauth

### LINKED ISSUE #10513: /api/authenticate/baseauth throws exception when auth header missing
`GET /api/authenticate/baseauth` should return an informative response

[example](https://usegalaxy.org/api/authenticate/baseauth)

Currently returns
```json
{"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
```

### PR DESCRIPTION
**Pull request by OlegZharkov**
resolves https://github.com/galaxyproject/galaxy/issues/10513

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/authenticate.py [hunk 1]
```diff
@@ -92,7 +92,15 @@ def _decode_baseauth(self, encoded_str):
...
-        split = encoded_str.strip().split(' ')
+        try:
+            split = encoded_str.strip().split(' ')
+        except AttributeError:
+            if not encoded_str:
+                raise exceptions.MessageException(
+                    'Authentication is missing')
+            else:
+                raise exceptions.MessageException(
+                    'Failed to authenticate')
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/authenticate.py]: This is more readable if we don't wrap it so short, right?  We don't strictly adhere to 80 in the Galaxy codebase.

```suggestion
                raise exceptions.MessageException('Authentication is missing')
            else:
                raise exceptions.MessageException('Failed to authenticate')
```

File: lib/galaxy/webapps/galaxy/api/authenticate.py [hunk 2]
```diff
@@ -92,7 +92,13 @@ def _decode_baseauth(self, encoded_str):
 
         :raises: HTTPBadRequest
         """
-        split = encoded_str.strip().split(' ')
+        try:
+            split = encoded_str.strip().split(' ')
+        except AttributeError:
+            if not encoded_str:
+                raise exceptions.MessageException('Authentication is missing')
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/authenticate.py]: This should be the more specific RequestParameterInvalidException
```suggestion
                raise exceptions.RequestParameterInvalidException('Authentication is missing')
```

File: lib/galaxy/webapps/galaxy/api/authenticate.py [hunk 3]
```diff
@@ -92,7 +92,13 @@ def _decode_baseauth(self, encoded_str):
...
+            else:
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/authenticate.py]: In what other circumstances would this throw an AttributeError?

Comment (OlegZharkov) [reply lib/galaxy/webapps/galaxy/api/authenticate.py]: if it's called from another function on backend? I thought it won't be good to always return 'Authentication is missing', but I guess you're right. I will remove it.

Comment (dannon) [reply lib/galaxy/webapps/galaxy/api/authenticate.py]: Yeah, what I was getting at was how else would this throw an AttributeError other than encoded_str being missing.  Agreed, the single "RequestParameterInvalidException" is enough.

**Pull request merged**